packages/endo/src/parse.js

```diff
@@ -78,16 +78,21 @@ export const parseJson = (source, location) => {
   return freeze({ imports, execute });
 };
 
-export const makeExtensionParser = extensions => {
-  return (source, location) => {
-    const extension = parseExtension(location);
+export const makeExtensionParser = (extensions, types) => {
+  return (source, specifier, location, packageLocation) => {
+    let extension;
+    if (typeof types === "object" && hasOwnProperty.call(types, specifier)) {
```

Comment:
Prefer coercing to an object instead of `typeof`.

```suggestion
    if (Object(types) === types && hasOwnProperty.call(types, specifier)) {
```

Reply:
Oh, `null`, thou nave.